internal/adminapi/client.go

```diff
@@ -74,39 +71,3 @@ func MakeHTTPClient(opts *HTTPClientOpts) (*http.Client, error) {
 
 	return c, nil
 }
-
-// GetKongClientForWorkspace returns a Kong API client for a given root API URL and workspace.
-// If the workspace does not already exist, GetKongClientForWorkspace will create it.
-func GetKongClientForWorkspace(ctx context.Context, adminURL string, wsName string,
```

Comment:
question: I'm not opposed to removing this, but can you help me to understand why we removed this functionality instead of extending it for our needs in this PR?

Reply:
The removed implementation use non-admin rbac user token (when pass in through --kong-admin-token flag) access :/workspaces/{ws-name} interface, which is not allowed because only kong-admin user is able to access that endpoints.

We need kong-admin-token verify the existence of a non-default workspace, also need non-admin user token access the specified non-default workspace entities, which is fixed in the new function.

Reply:
@shaneutt  does my explanation help ?

Reply:
While I appreciate your explanation @ccfishk, this only answers why we made changes, but not why we moved the entire thing. If we have _no_ reasoning for moving the function then I would like to see us not move it, but simply extend it for this PR.

Reply:
reverted 5ac1d90c5e6aa8319c4fd3169d6e45da4810e7cf